core/datasource/external/ethverifier/verifier_snapshot.go

```diff
@@ -128,9 +128,11 @@ func (s *Verifier) LoadState(ctx context.Context, payload *types.Payload) ([]typ
 }
 
 func (s *Verifier) OnStateLoaded(ctx context.Context) error {
-	// tell the eth call engine what the last block seen was so it does not retrigger calls
+	// tell the eth call engine what the last block seen was, so it does not re-trigger calls
 	if s.lastBlock != nil && s.lastBlock.Height > 0 {
-		s.ethEngine.UpdatePreviousEthBlock(s.lastBlock.Height, s.lastBlock.Time)
+		s.ethEngine.StartAtHeight(s.lastBlock.Height, s.lastBlock.Time)
+	} else {
+		s.ethEngine.Start()
```

Comment:
Not entirely convinced this else clause is necessary, in what way would it be possible to load from a snapshot and have these values as the defaults?  seems improbable but not impossible